### ISSUE TITLE
Error when loaded via BTML v0.2.1 (causes infinite loading screen in Skirmish Mode's MechLab)

### ISSUE THREAD

**Issue by TimeDiver0** [CptMoore/DynModLib#6]
Bit of a cross-post with the main BTML repository, but here's the appropriate BTModLoader.log excerpt:
```
11:27:19 PM - DynModLib.dll: Found and called entry point with params: Main.Start
11:27:19 PM - DynModLib.dll: While loading a dll, an exception occured:
System.Reflection.TargetInvocationException: Exception has been thrown by the target of an invocation. ---> System.MissingMethodException: Method not found: 'Harmony.CollectionExtensions.Do'.
  at (wrapper managed-to-native) System.Reflection.MonoMethod:InternalInvoke (object,object[],System.Exception&)
  at System.Reflection.MonoMethod.Invoke (System.Object obj, BindingFlags invokeAttr, System.Reflection.Binder binder, System.Object[] parameters, System.Globalization.CultureInfo culture) [0x00000] in <filename unknown>:0 
  --- End of inner exception stack trace ---
  at System.Reflection.MonoMethod.Invoke (System.Object obj, BindingFlags invokeAttr, System.Reflection.Binder binder, System.Object[] parameters, System.Globalization.CultureInfo culture) [0x00000] in <filename unknown>:0 
  at System.Reflection.MethodBase.Invoke (System.Object obj, System.Object[] parameters) [0x00000] in <filename unknown>:0 
  at BattleTechModLoader.BTModLoader.LoadDLL (System.String path, System.String methodName, System.String typeName, System.Object[] prms, BindingFlags bFlags) [0x00000] in <filename unknown>:0 
```
Reverting to the 2 .dll(s) from BTML v0.2.0 gets rid of the error message and **_doesn't_** lock up the game.

**Comment by CptMoore** [comment]
thx I'll upgrade my code to the newest harmony version asap

**Comment by CptMoore** [comment]
I made an update of dynmodlib, does it work for you now? and is it backwards compatible?

**Comment by TimeDiver0** [comment]
BTModLoader.log no longer has above error, but has not resolved infinite load issue.

One other mod has a similar error; Commander Portrait Loader, may submit report later.

However, disabling that one mod by itself did **not** resolve the issue, either.

Rolling back to BTML v0.2.0; Harmony v1.1.0 does _something_ different.

**Comment by CptMoore** [comment]
Can you check then if the new dynmodlib works for btml 1.0? Or do we now need to provide 2 releases?

**Comment by TimeDiver0** [comment]
I briefly tried out the latest DynModLib with BTML v0.2.0's DLLs, and it didn't cause any freezes or anything, but that was just checking the MechBay for Skirmish.

Not sure how it would affect the tactical game layer, or for that matter re-compiling the updated source code for some of your other mods. Testing that now...

Addendum: the re-compiling of HardpointFixMod and StartingMercsMod via latest DynModLib w/ BTML v0.2.0's DLLs worked just fine, moving on to tactical layer testing.

**Comment by CptMoore** [comment]
I really appretiate you doing the testing, im at work and cant test myself

**Comment by CptMoore** [comment]
I have been able to fix BTML 0.2.1 or more precisely, harmony 1.1 issues properly with all my mods. now on to the new modtek which doesnt work nicely.
